src/lib.rs

```diff
@@ -224,6 +229,92 @@ pub mod pin {
         }
     }
 
+    /// A pin with software controlled PWM output.
+    pub struct SoftPwmPin<Pin>(libc::c_int, PhantomData<Pin>);
+
+    impl<P: Pin + RequiresRoot> SoftPwmPin<P> {
+        /// Creates a new software PWM controlled pin.
+        ///
+        /// Due to limitations of the chip only one pin is able to do
+        /// hardware-controlled PWM output. The `SoftPwmPin`s on the
+        /// other hand allow for all GPIOs to output PWM signals.
+        ///
+        /// The pulse width of the signal will be 100μs with a value range
+        /// of [0,100] \(where `0` is a constant low and `100` is a
+        /// constant high) resulting in a frequenzy of 100 Hz.
+        ///
+        /// **Important**: In order to use software PWM pins *wiringPi*
+        /// has to be setup in GPIO mode via `setup_gpio()`.
+        pub fn new(pin: libc::c_int) -> SoftPwmPin<P> {
+            unsafe {
+                bindings::softPwmCreate(pin, 0, 100);
+            }
+
+            SoftPwmPin(pin, PhantomData)
+        }
+
+        #[inline]
+        pub fn number(&self) -> libc::c_int {
+            let &SoftPwmPin(number, _) = self;
+            number
+        }
+
+        /// Sets the duty cycle.
+        ///
+        /// `value` has to be in the interval [0,100].
+        pub fn pwm_write(&self, value: libc::c_int) {
+            unsafe {
+                bindings::softPwmWrite(self.number(), value);
+            }
+        }
+
+        pub fn into_input(self) -> InputPin<P> {
+            let SoftPwmPin(number, _) = self;
+            unsafe {
+                bindings::softPwmStop(number);
+            }
+            InputPin::new(number)
+        }
+
+        pub fn into_output(self) -> OutputPin<P> {
+            let SoftPwmPin(number, _) = self;
+            unsafe {
+                bindings::softPwmStop(number);
+            }
+            OutputPin::new(number)
+        }
+
+    }
+
+    impl<P: Pin + Pwm> SoftPwmPin<P> {
+        pub fn into_pwm(self) -> PwmPin<P> {
+            let SoftPwmPin(number, _) = self;
+            unsafe {
+                bindings::softPwmStop(number);
+            }
+            PwmPin::new(number)
+        }
+    }
+
+    impl<P: Pin + GpioClock> SoftPwmPin<P> {
+        pub fn into_clock(self) -> ClockPin<P> {
+            let SoftPwmPin(number, _) = self;
+            unsafe {
+                bindings::softPwmStop(number);
+            }
+            ClockPin::new(number)
+        }
+    }
+
+    impl<P> Drop for SoftPwmPin<P> {
```

Comment:
I'm not sure how good idea it is to implement `Drop`, since there are no guarantees that this is the only reference to this pin, even if it's discouraged. I'm also thinking that it may also be a bit surprising if they just "stop working" when it drops out of scope.